### ISSUE TITLE
Create Realm Button Missing

### ISSUE THREAD

**Issue by bcook-konza** [keycloak/keycloak#35115]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

Within the keycloak UI the 'Create Realm' button is missing from the drop-down on the top left of the UI

similar issue to issue described here in discourse group:
https://keycloak.discourse.group/t/keycloak-doesnt-allow-to-create-new-realm-no-button/2109

![image](https://github.com/user-attachments/assets/70afe0d5-7349-49a0-86ba-fb08e9a666b6)

### Version

26.0.5

### Regression

- [ ] The issue is a regression

### Expected behavior

For the UI realm drop-down to show the 'Create Realm' button

### Actual behavior

It only shows the existing realms, but does not display the 'Create Realm' button

### How to Reproduce?

This is a static consistent state within our environment

**Comment by jonkoops** [comment]
These are not steps to reproduce. We cannot magically infer your environment and solve this issue. Very likely you are using a user that does not have correct permissions to be able to create a realm. Please provide detailed steps to reproduce this issue locally from a clean Keycloak installation using the latest version.

**Comment by bcook-konza** [comment]
Thanks for the comment @jonkoops, we are a bit new to Keycloak and standing up our first Keycloak environment - to provide a bit more context this is running on an azure kubernetes pod with an azure postgres database. We seem to have all of the permissions required (including the original admin user). We were previously able to create realms early on in the deployment but the ability to do so seemed to have disappeared somewhere along the way. We will continue investigating on what the root cause is and try to define how to consistently reproduce the issue.

**Comment by ShaizanKhan** [comment]
I have the same issue. I don't see the Create Realm button as well. After deploying Keycloak, I created a new admin user with all the roles as that of bootstrapped admin user and deleted the default admin. 

![image](https://github.com/user-attachments/assets/9ae52941-55ab-4483-8596-f18973cb7bba)

**Comment by bcook-konza** [comment]
@jonkoops yeah it looks like the steps to reproduce here is deleting the bootstrapped admin - even if other admins exist with the same permissions. Not sure if this is the expected behavior

We were tightening down our users/logins and we may have wrongly assumed that the bootstrapped admin wasn't needed once we had all of our individual admin accounts setup

**Comment by jonkoops** [comment]
Go it, thanks for providing some more details, that really helps. Likely some additional steps need to be taken in terms of the user being qualified as an administrator that can create realms, I'll talk to the team to get to the bottom of this. We're also working on improving the administration bootstrapping under #33965, so it might be good to subscribe to that issue as well.

**Comment by areYouLazy** [comment]
26.0.6, same behavior

**Comment by mposolda** [comment]
I am not able to reproduce with the bare-metal Keycloak installation (ZIP distribution). What I did exactly is this:

1) Started keycloak on bare metal `./kc.sh start-dev`

2) Go to http://localhost:8080 and created user `admin`/`admin` on welcome screen

3) Login to the admin console as `admin` and created user `admin2`

4) Assigned password to `admin2` and realm role `admin` (This is the composite role, which gives the user `admin2` every other role and hence full control of admin console. More about admin permissions in the docs: https://www.keycloak.org/docs/latest/server_admin/index.html#_admin_permissions )

5) Logout and login as `admin2`

6) Deleted the original user `admin` created during bootstrap

7) Logout from `admin2` and restarted the Keycloak server (this step is not needed, but did just in case)

8) After restart, login again as `admin2` user. I still have full permission and I am able to create realms


Could you try these steps on bare-metal and check if it works for you? If things work as expected on bare-metal, but don't work on the kubernetes like setups, it is possible that kubernetes bootstrap is doing something during server bootstrap, which makes things to not work as expected. It is possible that kubernetes bootstrap adds some environment variables at the start or something else. In this case, this would be probably the issue for cloud-native team.

AFAIK there were some changes recently in the admin bootstrap, which are maybe related to this issue (not 100% sure): https://www.keycloak.org/server/bootstrap-admin-recovery

**Comment by areYouLazy** [comment]
Thanks @mposolda I think the problem is that while on the "temporary" admin account and creating the "real" admin account, I'm not able to assign the role `admin` as I cannot see it in the `role mapping` page
There's seems to be no `admin` role, or I'm missing where to assign it

![image](https://github.com/user-attachments/assets/abc623e0-157f-4e7e-b978-d47f59714473)

I'm not able to test the bare-metal installation but I am incline to trust you on your findings :)

**Comment by mposolda** [comment]
@areYouLazy The role `admin` is visible in `Filter by realm roles` (the very first combobox). Are you not seeing there? See here for some more info in the role mappings: https://www.keycloak.org/docs/latest/server_admin/index.html#_admin_permissions .

If it is the only issue, maybe we should somehow improve documentation (or admin console usability ) as realm roles are maybe bit hidden (due the fact that "Filter by clients" is shown by default), but not sure...

**Comment by areYouLazy** [comment]
Aaaaa I see! You're right!!!

Ok that seems a documentation problem, but one thing that comes to my mind is: Is the 1st administrator always marked as temporary, even if you set a strong password on it? If the answer is yes, I think we should improve the documentation as you will always need to generate a new administrator from the web interface.

Another thing to consider would be to prevent the deletion of the temporary admin account until you get the admin role on another user (at least), but this will end up in a bunch of github issues if we still don't improve documentation

**Comment by jonkoops** [comment]
Considering this works, but is simply a case of poor documentation/UX I will close this issue in favor of handling this under #33965

**Comment by godsey** [comment]
I had deleted the temporary admin and couldn't seem to get the admin role to show per above. I used this query to give myself all roles. I had to restart the container for it to pickup changes.

INSERT INTO user_role_mapping (role_id, user_id)
SELECT 
    kr.id AS role_id, 
    ue.id AS user_id
FROM 
    keycloak_role kr,
    (SELECT id FROM user_entity WHERE username = 'binutils@email.com' LIMIT 1) ue
ON CONFLICT DO NOTHING;

**Comment by jackofsometrade** [comment]
>     4. Assigned password to `admin2` and realm role `admin` (This is the composite role, which gives the user `admin2` every other role and hence full control of admin console. More about admin permissions in the docs: https://www.keycloak.org/docs/latest/server_admin/index.html#_admin_permissions )

I have a fresh install of `v26.1.0` and there is no `admin` realm role. Really struggling here. That doesn't seem to be a known outcome. Reproduce by simply `docker run quay.io/keycloak/keycloak:26.1.0`. Log in with temporary admin account, create a new user and try to find an `admin` role, it won't show up.

**Comment by MABV-NNIT** [comment]
I know I'm late to the party @jackofsometrade.. But for those who are ending on this page due to not seeing the roles or the option for filtering. 
It's supposed to look like this (to give admin rights select the "filter by realm roles")

<img width="1105" height="196" alt="Image" src="https://github.com/user-attachments/assets/8ebea06a-029a-4567-89f7-3b7b6e1fd7a4" />

However if you're seeing this and only this (Most likely you've selected all available roles and assigned them to the user): 
<img width="1120" height="432" alt="Image" src="https://github.com/user-attachments/assets/99c16b53-2111-46f5-9107-6a9371d1a5fa" />
The way that I found the option to go back and see the first option is to remove the assigned roles from an admin user, and then it reappears. 

I was able to reproduce this for the user "test" by simply assigning all roles from the view "Filter by clients" - clicking the button "filter by clients" does not change the view for me.
